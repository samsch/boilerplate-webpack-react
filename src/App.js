import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';

const App = props =>
  <div>
    <h1>Counter Example</h1>
    <div>
      Current count: {props.state.count}
    </div>
    <div>
      <button type="button" onClick={props.action.increment}>
        Increment
      </button>{' '}
      <button type="button" onClick={props.action.decrement}>
        Decrement
      </button>
    </div>
    <div>
      <button type="button" onClick={props.action.reset}>
        Reset
      </button>
    </div>
  </div>;
App.propTypes = {
  state: PropTypes.shape({
    count: PropTypes.number.isRequired,
  }),
  action: PropTypes.shape({
    increment: PropTypes.func.isRequired,
    decrement: PropTypes.func.isRequired,
    reset: PropTypes.func.isRequired,
  }),
};

export default App;
