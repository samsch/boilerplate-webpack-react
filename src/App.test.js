/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('The App component', () => {
  test('Should match a snapshot render', () => {
    const props = {
      state: {
        count: 17,
      },
      action: {
        increment: jest.fn(),
        decrement: jest.fn(),
        reset: jest.fn(),
      },
    };

    const renderedApp = renderer.create(<App {...props} />);
    expect(renderedApp.toJSON()).toMatchSnapshot();
  });
});
