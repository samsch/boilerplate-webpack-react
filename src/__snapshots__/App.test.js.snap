// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component Should match a snapshot render 1`] = `
<div>
  <h1>
    Counter Example
  </h1>
  <div>
    Current count: 
    17
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Increment
    </button>
     
    <button
      onClick={[Function]}
      type="button"
    >
      Decrement
    </button>
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
